e2e/cypress/integration/account_settings/general/email_spec.js

```diff
@@ -48,4 +59,163 @@ describe('Account Settings -> General -> Email', () => {
+        cy.get('#currentPassword').should('be.visible').type("randompass");
+
+        // # Save the settings
+        cy.get('#saveSetting').click().wait(TIMEOUTS.HALF_SEC);
+
+        // * Check that the correct error message is shown.
+        cy.get('#clientError').should('be.visible').should('have.text', 'The new emails you entered do not match.');
+    });
+
+    // This test is a combination of 2 sub-tests because you cannot really complete 2069 without doing 2070.
+    // Doing them individually would have a dependency with the previous test.
+    // Hence, a combined single test for everything is better.
+    it('MM-T2069/MM-T2070 email address verification', () => {
+        // # Click "Edit" to the right of "Email"
+        cy.get('#emailEdit').should('be.visible').click();
+
+        const randomId = getRandomId();
+        // # Type new email
+        cy.get('#primaryEmail').should('be.visible').type(`user-${randomId}@example.com`);
+        cy.get('#confirmEmail').should('be.visible').type(`user-${randomId}@example.com`);
+        cy.get('#currentPassword').should('be.visible').type(testUser.password);
+
+        // # Save the settings
+        cy.get('#saveSetting').click().wait(TIMEOUTS.HALF_SEC);
+
+        // * Verify the announcement bar
+        cy.get('.announcement-bar').should('be.visible').should('contain.text', 'Check your email inbox to verify the address.');
+
+        const mailUrl = getEmailUrl(Cypress.config('baseUrl'));
+
+        cy.task('getRecentEmail', {username: `user-${randomId}`, mailUrl}).then((response) => {
+            // * Verify the subject
+            expect(response.data.subject).to.equal('[Mattermost] Verify new email address');
+
+            const bodyText = response.data.body.text.split('\n');
+
+            // * Verify the body
+            expect(bodyText[1]).to.contain('You updated your email');
+            const matched = bodyText[6].match(reUrl);
+            assert(matched.length > 0);
+
+            const permalink = matched[0];
+            // # Click on the link
+            cy.visit(permalink);
+
+            // * Verify announcement bar
+            cy.get('.announcement-bar').should('be.visible').should('contain.text', 'Email verified');
+        });
+
+        // # Wait for one second for the mail to be sent out.
+        cy.wait(TIMEOUTS.ONE_SEC);
+
+        cy.task('getRecentEmail', {username: testUser.username, mailUrl}).then((response) => {
+            // * Verify the subject
+            expect(response.data.subject).to.equal('[Mattermost] Your email address has changed');
+        });
+
+
+        cy.toAccountSettingsModal();
+
+        // * Verify new email address (MM-2070)
+        cy.get('#emailDesc').should('be.visible').should('have.text', `user-${randomId}@example.com`);
+    });
+
+
+    // it('MM-T2071 Change email address: update email verification prompt ', () => {
+
+    // });
+
+    // it('MM-T2072', () => {
+
+    // });
```

Comment:
These test cases are incorrect or already covered by other tests. I have raised a question in the ~UI Test Automation channel.